admin/ops/setup_dev_bucket.py

```diff
@@ -0,0 +1,340 @@
+import boto3
+import importlib
+import os
+from pathlib import Path
+import typer
+
+PROD_RECIPES_BUCKET = "edm-recipes"
+PROD_PUBLISHING_BUCKET = "edm-publishing"
+os.environ["RECIPES_BUCKET"] = PROD_RECIPES_BUCKET
+os.environ["PUBLISHING_BUCKET"] = PROD_PUBLISHING_BUCKET
+
+from dcpy import configuration
+from dcpy.models.connectors.edm.recipes import DatasetKey
+from dcpy.models.connectors.edm.publishing import (
+    ProductKey,
+    BuildKey,
+    DraftKey,
+    PublishKey,
+)
+from dcpy.utils import s3
+from dcpy.utils.logging import logger
+from dcpy.connectors.edm import recipes, publishing
+from dcpy.lifecycle.builds import plan
+
+
+ROOT_PATH = Path(__file__).parent.parent
+BUCKET_PREFIX = "de-dev-"
+PRODUCTS = [
+    "db-cbbr",
+    "db-colp",
+    "db-cpdb",
+    "db-developments",
+    "db-facilities",
+    "db-green_fast_track",
+    "db-template",
+    "db-pluto",
+    "db-zoningtaxlots",
+]
+
+
+def copy_folder(
+    source_bucket: str,
+    target_bucket: str,
+    path: str,
+) -> None:
+    """Copies folder from a production bucket to a dev bucket"""
+    assert target_bucket.startswith(BUCKET_PREFIX)
+    logger.info(f"Copying {source_bucket}/{path} to ./tmp")
+    s3.download_folder(
+        source_bucket, prefix=path, export_path=Path("./tmp/dev_bucket_setup")
+    )
+    logger.info(f"Copying ./tmp to {target_bucket}/{path}")
+    s3.upload_folder(
+        target_bucket,
+        Path(f"./tmp/dev_bucket_setup/{path}"),
+        Path(path),
+        "private",
+        contents_only=True,
+    )
+
+
+def resolve_latest_recipe(
+    target_bucket: str,
+    ds: str,
+):
+    """
+    Within a dev bucket, resolves/copies the dataset currently in "latest"
+    into its versioned folder under "datasets/{dataset_id}"
+    """
+    os.environ["RECIPES_BUCKET"] = target_bucket
+    importlib.reload(configuration)
+    assert PROD_RECIPES_BUCKET != PROD_RECIPES_BUCKET
```

Comment:
I'm guessing this should be configuration.RECIPES_BUCKET

Reply:
Fixed in a later commit - I would just look at the whole PR

Reply:
oh sorry. I was going one by one